### PR TITLE
fix xfs fail, add upload result when timeout

### PR DESCRIPTION
**Pull request by paxue**
increase the core number to 8 for xfs verify_generic_standard_datadisk
add try logic for exception to upload the test result
reduce the wait timeout in wait by 20s to let wat timeout happen before the whole case timeout.

### CODE REVIEW

File: microsoft/testsuites/xfstests/xfstesting.py [hunk 1]
```diff
@@ -534,7 +535,7 @@ def _execute_xfstests(
             mount_opts,
         )
         xfstests.set_excluded_tests(excluded_tests)
-        xfstests.run_test(test_type, log_path, result, data_disk, self.TIME_OUT)
+        xfstests.run_test(test_type, log_path, result, data_disk, self.TIME_OUT - 20)
```
Comment (squirrelsc) [inline microsoft/testsuites/xfstests/xfstesting.py]: Add comments above to explain the magic number.

File: microsoft/testsuites/xfstests/xfstests.py [hunk 2]
```diff
@@ -171,11 +171,15 @@ def run_test(
...
         # this is the actual process name, when xfstests runs.
-        pgrep.wait_processes("check", timeout=timeout)
-
-        self.check_test_results(
-            log_path=log_path, test_type=test_type, result=result, data_disk=data_disk
-        )
+        try:
+            pgrep.wait_processes("check", timeout=timeout)
+        finally:
+            self.check_test_results(
```
Comment (squirrelsc) [inline microsoft/testsuites/xfstests/xfstests.py:177]: Make sure the `check_test_results` won't raise another exception by incomplete content.

File: microsoft/testsuites/xfstests/xfstesting.py (hunk 1 above)
Comment (paxue) [reply microsoft/testsuites/xfstests/xfstesting.py]: Sure. I will add it

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

File: microsoft/testsuites/xfstests/xfstesting.py [hunk 3]
```diff
@@ -144,6 +144,7 @@ def before_case(self, log: Logger, **kwargs: Any) -> None:
          standard data disk with xfs type system.
         """,
         requirement=simple_requirement(
+            min_core_count=8,
```
Comment (LiliDeng) [inline microsoft/testsuites/xfstests/xfstesting.py]: there is no enough evidence to show the timeout issue is related with core count, let's remove this change, and keep monitor the test results data, then decide the next step.

Comment (paxue) [reply microsoft/testsuites/xfstests/xfstesting.py]: removed

**Review by LiliDeng — APPROVED** [review]

File: microsoft/testsuites/xfstests/xfstesting.py [hunk 4]
```diff
@@ -125,7 +125,7 @@ class Xfstesting(TestSuite):
     # commit d0c7feaf8767 ("xfs: add agf freeblocks verify in xfs_agf_verify")
     # TODO: will figure out the detailed reason of every excluded case.
     excluded_tests = (
-        "generic/211 generic/430 generic/431 generic/434 xfs/438 xfs/490"
+        "generic/211 generic/430 generic/431 generic/434 generic/738 xfs/438 xfs/490"
```
Comment (squirrelsc) [inline microsoft/testsuites/xfstests/xfstesting.py:128]: Add comments above, why 738 is removed. Is the timeout resoled after this test is removed?

Comment (paxue) [reply microsoft/testsuites/xfstests/xfstesting.py:128]: yes, tested for many times, it seems 738 is the blocking issue. I will send another PR to add this comment. After removing 738 it passed on 2 core VMs

**Pull request merged**